src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -1 +1,132 @@
+                profile => profile.molecularProfileId
+            );
+            GenericAssayMetaGroupByGenericAssayType[
+                genericAssayType
+            ] = await fetchGenericAssayMetaByProfileIds(genericAssayProfileIds);
+        })
+    );
+    return GenericAssayMetaGroupByGenericAssayType;
+}
+
+export async function fetchGenericAssayMetaByMolecularProfileIdsGroupByMolecularProfileId(
+    molecularProfiles: MolecularProfile[]
+) {
+    const genericAssayProfiles = molecularProfiles.filter(profile => {
+        return (
+            profile.molecularAlterationType ===
+            AlterationTypeConstants.GENERIC_ASSAY
+        );
+    });
+
+    const genericAssayMetaGroupByMolecularProfileId: {
+        [molecularProfileId: string]: GenericAssayMeta[];
+    } = {};
+
+    await Promise.all(
+        genericAssayProfiles.map(async genericAssayProfile => {
+            const genericAssayProfileId =
+                genericAssayProfile.molecularProfileId;
+            genericAssayMetaGroupByMolecularProfileId[
+                genericAssayProfileId
+            ] = await fetchGenericAssayMetaByProfileIds([
+                genericAssayProfileId,
+            ]);
+        })
+    );
+    return genericAssayMetaGroupByMolecularProfileId;
+}
+
+export async function fetchGenericAssayMetaByProfileIds(
+    genericAssayProfileIds: string[]
+) {
+    let genericAssayMeta: GenericAssayMeta[] = [];
+    if (genericAssayProfileIds.length > 0) {
+        genericAssayMeta = await client.fetchGenericAssayMetaDataUsingPOST({
+            genericAssayMetaFilter: {
+                molecularProfileIds: genericAssayProfileIds,
+                // the Swagger-generated type expected by the client method below
+                // incorrectly requires both molecularProfileIds and genericAssayStableIds;
+                // use 'as' to tell TypeScript that this object really does fit.
+            } as GenericAssayMetaFilter,
+        });
+    }
+    return genericAssayMeta;
+}
+
+export async function fetchGenericAssayMetaByEntityIds(entityIds: string[]) {
+    let genericAssayMeta: GenericAssayMeta[] = [];
+    if (entityIds.length > 0) {
+        genericAssayMeta = await client.fetchGenericAssayMetaDataUsingPOST({
+            genericAssayMetaFilter: {
+                genericAssayStableIds: entityIds,
+                // the Swagger-generated type expected by the client method below
+                // incorrectly requires both molecularProfileIds and genericAssayStableIds;
+                // use 'as' to tell TypeScript that this object really does fit.
+            } as GenericAssayMetaFilter,
+        });
+    }
+    return genericAssayMeta;
+}
+
+export async function fetchGenericAssayData(
+    entityIdsByProfile: { [molecularProfileId: string]: string[] },
+    sampleFilterByProfile: { [molecularProfileId: string]: IDataQueryFilter }
+) {
+    const params = Object.keys(entityIdsByProfile).map(profileId => ({
```

Comment:
you can use _.map on an object and it will iterate through keys and give you both key and item.  it will result in an array.

Reply:
@alisman I can not use _.map on an object. https://lodash.com/docs/4.17.15#map. The _.map can only be used on collection. So I will change it to chain, maybe that can make this better.

Reply:
@dippindots you can definitely use map on an object, if you look closer the collection can be an Array or an Object. if you test it in the console you can see how it works!

Reply:
Thanks, I will change it to map!